### PR TITLE
Styling for geojson feature

### PR DESCRIPTION
**Pull request by thiloSchlemmer**
Vector-Features which are imported from geojson-resources haven't a style yet.
This branch tries to solve this, by implementing a style-mapping according to the [Mapbox Simple Style Specification v1.1.0](https://github.com/mapbox/simplestyle-spec/tree/master/1.1.0).

### CODE REVIEW

**Comment by thiloSchlemmer** [comment]
simpleStyleExample.geojson:
```
{
	"type" : "FeatureCollection",
	"name" : "Simple Style Example",
	"features" : [
		{
			"type" : "Feature",
			"geometry" : {
				"type" : "Polygon",
				"coordinates" : [
					[
						[ 11.6083928702, 48.0996233417 ],
						[ 11.6095284133, 48.09969915 ],
						[ 11.6096043253, 48.1003662667 ],
						[ 11.6096412992, 48.1005377207 ],
						[ 11.6097323167, 48.100788767 ],
						[ 11.6097922865, 48.1009010223 ],
						[ 11.6098512332, 48.100989627 ],
						[ 11.6098302912, 48.1011134929 ],
						[ 11.6098038364, 48.1012019869 ],
						[ 11.6097091988, 48.1013051764 ],
						[ 11.6094900014, 48.1012812287 ],
						[ 11.6092366424, 48.1012631046 ],
						[ 11.6082875483, 48.1011878367 ],
						[ 11.6064409111, 48.1010587824 ],
						[ 11.6064712675, 48.1009697598 ],
						[ 11.6064894962, 48.1008866362 ],
						[ 11.6065150868, 48.1007520028 ],
						[ 11.6065552166, 48.1005247825 ],
						[ 11.606657097, 48.0997892928 ],
						[ 11.6066867622, 48.0995597006 ],
						[ 11.6083928702, 48.0996233417 ]
					]
				]
			},
			"properties" : {
                "title": "Polygon 1",
                "description": "A valid Polygon style",
                "stroke": "#ff00ee",
                "stroke-opacity": 1.0,
                "stroke-width": 1.0,   
                "fill": "#ff00ee",            
                "fill-opacity": 0.5   
			}
		},
		{
			"type" : "Feature",
			"geometry" : {
				"type" : "Polygon",
				"coordinates" : [
					[
						[ 11.614926084, 48.1013600695 ],
						[ 11.6144908232, 48.1014095169 ],
						[ 11.6139281567, 48.1014941476 ],
						[ 11.613932737, 48.1012656193 ],
						[ 11.6140314448, 48.1012624188 ],
						[ 11.6140804055, 48.1012612773 ],
						[ 11.6164428237, 48.1011702712 ],
						[ 11.6164378929, 48.1013337429 ],
						[ 11.6153540515, 48.1013323638 ],
						[ 11.614926084, 48.1013600695 ]
					]
				]
			},
			"properties" : {
				"title": "Polygon 2",
                "description": "A incomplete but valid Polygon style",
                "stroke": "#ff00ee",
                "stroke-opacity": 1.0
			}
		},
		{
			"type" : "Feature",
			"geometry" : {
				"type" : "LineString",
				"coordinates" : [
						[ 11.623994925, 48.103902276 ],
						[ 11.6238941494, 48.1038562591 ],
						[ 11.6237933577, 48.1038312889 ],
						[ 11.623671698, 48.1038326017 ],
						[ 11.6236550072, 48.103838031 ],
						[ 11.6234912923, 48.1034946718 ],
						[ 11.6234954066, 48.1030211821 ],
						[ 11.6240959829, 48.1030061469 ],
						[ 11.6240767886, 48.1032433238 ],
						[ 11.6252867619, 48.1032918834 ]
				]
			},
			"properties" : {				
				"title": "LineString 1",
                "description": "A valid LineString style",
                "stroke": "#00ffee",
                "stroke-opacity": 1.0
			}
		},
		{
			"type" : "Feature",
			"geometry" : {
				"type" : "Point",
				"coordinates" : 
						[ 11.6160251361, 48.1052634623 ]
			},
			"properties" : {				
				"title": "Point 1",
                "description": "A valid Point style",
                "marker-size": "medium",
                "marker-symbol": "rocket",
                "marker-color": "#00f"
			}
		}
	]
}
```

**Comment by thiloSchlemmer** [comment]
A different approach, using features of CSS, was specified by [Geojson CSS](https://wiki.openstreetmap.org/wiki/Geojson_CSS).
This styling tries to use CSS-Elements in `style`-Nodes on Feature- or FeatureCollection-Level.

File: src/modules/map/components/olMap/olStyleUtils.js [hunk 1]
```diff
@@ -112,6 +110,102 @@ export const getMarkerSrc = (symbolSrc = null, symbolColor = '#ffffff') => {
...
+		switch (markerSize) {
+			case 'large':
+				return 7;
+			case 'medium':
+				return 5;
+			case 'small':
+			default:
+				return 3;
+		}
+	};
```
Comment (taulinger) [inline src/modules/map/components/olMap/olStyleUtils.js:180]: not covered by a test

File: src/modules/map/components/olMap/olStyleUtils.js [hunk 2]
```diff
@@ -112,6 +110,102 @@ export const getMarkerSrc = (symbolSrc = null, symbolColor = '#ffffff') => {
...
+
+	const featureStyleProperties = {};
+	Object.keys(defaultStyleProperties).forEach(k => {
+		const styleValue = feature.get(k);
+		if (styleValue) {
+			featureStyleProperties[k] = styleValue;
+		}
+	});
```
Comment (taulinger) [inline src/modules/map/components/olMap/olStyleUtils.js:188]: not covered by a test

**Review by taulinger — CHANGES_REQUESTED** [review]
Seems there're missing some tests to have a better test coverage (https://coveralls.io/builds/47327314)

**Review by taulinger — APPROVED** [review]
Thanks!

**Pull request merged**